library_generation/generate_composed_library.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+# This script allows generation of libraries that are composed of more than one
+# service version. It is achieved by calling `generate_library.sh` without
+# postprocessing for all service versions and then calling
+# postprocess_library.sh at the end, once all libraries are ready.
+#
+# Arguments
+# --generation_queries: a single string of comma-separated key-value groups separated by a
+# pipe | (i.e. the groups are spearated by pipe, while a group's key-values are
+# separated by comma). They key-value groups are in the form of `key=value` and will
+# be converted to an argument to generate_library.sh (`--key value`).
+#   example: "proto_path=google/cloud/asset/v1,destination_path=google-cloud-asset-v1-java,(...)|proto_path=google/cloud/asset/v1p2beta5,destination_path=google-cloud-asset-v1-java,(...)"
```

Comment:
So the caller of this script will have to prepare this long query, which is not very intuitive. Is it possible to convert this to a Python script? So that we can pass a model class that better represents the parameters? For example, I'm expecting a Python script that parses the yaml config to a parent model class, and pass the sub-models to generate_composed_library.py.
I see that we are using Python utils anyway and only need the sparse_clone from shell utils, but we may not need it per my comment below.

Reply:
> the caller of this script will have to prepare this long query, which is not very intuitive.

This is a great question. I'm feeling that I'm not familiar with how we will operate around this script.

@diegomarquezp  Am I right that humans do not construct the argument?

@diegomarquezp  @blakeli0  If you prepare a hypothetical procedure document when and how to use this `generate_composed_library.sh`, that would answer the questions and optimal way to structure multiple scripts.

Reply:
> do humans construct the argument?
@suztomo at the moment, the arguments are prepared programatically in the integration test, but as @blakeli0 pointed out, it may be convenient to convert this to a python script so a higher-level main script could parse a configuration yaml and call the script as a function instead of passing a serialized string

Reply:
Right, humans are not expected to construct the arguments, it is being abstracted away by the yaml config file. We have a [scratch doc](https://docs.google.com/document/d/1O457OrQAy_JT7cf-5kp_kgnPvfOuiMo9OLJH5m0rIF4/edit?resourcekey=0-uzkHb7jXtXezlb-qk_HeoQ&tab=t.0) that illustrated the high level flow of, we should probably formalize it.